src/Reactant.jl

```diff
@@ -8,6 +8,7 @@ include("OrderedIdDict.jl")
 using Enzyme
 
 abstract type RArray{T,N} <: AbstractArray{T,N} end
+abstract type RScalar{T} <: Number end
```

Comment:
maybe let's call it `RNumber` to be as similar to Julia as posible?
```suggestion
abstract type RNumber{T} <: Number end
```